client/client-multi/client-c/src/kaa/kaa_notification_manager.c

```diff
@@ -0,0 +1,1035 @@
+                        uid_length = KAA_NTOHS(*((uint16_t *) reader->current));
+                        reader->current += sizeof(uint16_t);
+                        extension_length -= sizeof(uint16_t);
+                        notification_size = KAA_NTOHL(*((uint32_t *) reader->current));
+                        reader->current += sizeof(uint32_t);
+                        extension_length -= sizeof(uint32_t);
+                        topic_id =  KAA_NTOHLL(*((uint64_t *) reader->current));
+                        reader->current += sizeof(uint64_t);
+                        extension_length -= sizeof(uint64_t);
+                        KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Topic id is %lu", topic_id);
+                        err = kaa_find_topic(self, &topic_found, &topic_id);
+                        if (err) {
+                            reader->current += kaa_aligned_size_get(uid_length);
+                            reader->current += kaa_aligned_size_get(notification_size);
+                            continue;
+                        }
+                        if (uid_length) {
+                            kaa_notifications_uid_t *uid = (kaa_notifications_uid_t *) KAA_MALLOC(sizeof(kaa_notifications_uid_t));
+                            KAA_RETURN_IF_NIL(uid, KAA_ERR_NOMEM);
+                            uid->length = uid_length;
+                            uid->data = KAA_MALLOC(uid_length);
+                            if (!uid->data) {
+                                KAA_FREE(uid);
+                                return KAA_ERR_NOMEM;
+                            }
+                            err = kaa_platform_message_read_aligned(reader, uid->data, uid->length);
+                            if (err) {
+                                KAA_LOG_WARN(self->logger, KAA_ERR_BADDATA, "Failed to read UID body");
+                            }
+                            extension_length -= kaa_aligned_size_get(uid->length);
+                            kaa_list_t * new_uid = NULL;
+                            if (!self->uids) {
+                                new_uid = kaa_list_create(uid);
+                            } else {
+                                new_uid = kaa_list_push_front(self->uids, uid);
+                            }
+                            if (!new_uid) {
+                                KAA_FREE(uid->data);
+                                KAA_FREE(uid);
+                                return KAA_ERR_NOMEM;
+                            }
+                            self->uids = new_uid;
+                        }
+                        if (notification_size) {
+                            avro_reader_t avro_reader = avro_reader_memory(reader->current, notification_size);
+                            if (!avro_reader) {
+                                return KAA_ERR_NOMEM;
+                            }
+                            notification = KAA_NOTIFICATION_DESERIALIZE(avro_reader);
```

Comment:
Less code:
notification = KAA_NOTIFICATION_DESERIALIZE(avro_reader);
avro_reader_free(avro_reader);
if (!notification) {
    KAA_LOG_ERROR(self->logger, KAA_ERR_NONE, "Failed to deserialize notification.");
    return KAA_ERR_NOMEM;
}